src/map/utils/trustutils.cpp

```diff
@@ -239,197 +240,115 @@ void FreeTrustList()
-              (grade::GetHPScale(grade, scaleOver30Column) * mainLevelOver30) +
-              (grade::GetHPScale(grade, scaleOver60Column) * mainLevelOver60To75) +
-              (grade::GetHPScale(grade, scaleOver75Column) * mainLevelOver75);
-
-    // Расчет бонусных HP
-    bonusStat = (mainLevelOver10 + mainLevelOver50andUnder60) * 2;
-
-    // Расчет по support job
-    if (slvl > 0)
-    {
-        grade = grade::GetJobGrade(sjob, 0);
-
-        sJobStat = grade::GetHPScale(grade, baseValueColumn) +
-                   (grade::GetHPScale(grade, scaleTo60Column) * (slvl - 1)) +
-                   (grade::GetHPScale(grade, scaleOver30Column) * subLevelOver30) +
-                   subLevelOver30 + subLevelOver10;
-        sJobStat = sJobStat / 2;
-    }
-
-    PTrust->health.maxhp = (int16)(map_config.player_hp_multiplier * (raceStat + jobStat + bonusStat + sJobStat));
-
-    //Начало расчера MP
-
-    raceStat = 0;
-    jobStat = 0;
-    sJobStat = 0;
-
-    // Расчет MP расе.
-    grade = grade::GetRaceGrades(race, 1);
-
-    //Если у main job нет МП рейтинга, расчитиваем расовый бонус на основе уровня subjob уровня(при условии, что у него есть МП рейтинг)
-    if (grade::GetJobGrade(mjob, 1) == 0)
+    JOBTYPE mJob = PTrust->GetMJob();
+    JOBTYPE sJob = PTrust->GetSJob();
+    uint8 mLvl = PTrust->GetMLevel();
+
+    float growth = 1.06f;
+    float base = 18.0f;
+
+    PTrust->health.maxhp = (int16)(base * pow(mLvl, growth) * PTrust->HPscale);
+    PTrust->health.maxmp = (int16)(base * pow(mLvl, growth) * PTrust->MPscale);
+
+    ((CItemWeapon*)PTrust->m_Weapons[SLOT_MAIN])->setDamage(mobutils::GetWeaponDamage(PTrust));
+
+    uint16 fSTR = mobutils::GetBaseToRank(PTrust->strRank, mLvl);
+    uint16 fDEX = mobutils::GetBaseToRank(PTrust->dexRank, mLvl);
+    uint16 fVIT = mobutils::GetBaseToRank(PTrust->vitRank, mLvl);
+    uint16 fAGI = mobutils::GetBaseToRank(PTrust->agiRank, mLvl);
+    uint16 fINT = mobutils::GetBaseToRank(PTrust->intRank, mLvl);
+    uint16 fMND = mobutils::GetBaseToRank(PTrust->mndRank, mLvl);
+    uint16 fCHR = mobutils::GetBaseToRank(PTrust->chrRank, mLvl);
+
+    uint16 mSTR = mobutils::GetBaseToRank(grade::GetJobGrade(PTrust->GetMJob(), 2), mLvl);
+    uint16 mDEX = mobutils::GetBaseToRank(grade::GetJobGrade(PTrust->GetMJob(), 3), mLvl);
+    uint16 mVIT = mobutils::GetBaseToRank(grade::GetJobGrade(PTrust->GetMJob(), 4), mLvl);
+    uint16 mAGI = mobutils::GetBaseToRank(grade::GetJobGrade(PTrust->GetMJob(), 5), mLvl);
+    uint16 mINT = mobutils::GetBaseToRank(grade::GetJobGrade(PTrust->GetMJob(), 6), mLvl);
+    uint16 mMND = mobutils::GetBaseToRank(grade::GetJobGrade(PTrust->GetMJob(), 7), mLvl);
+    uint16 mCHR = mobutils::GetBaseToRank(grade::GetJobGrade(PTrust->GetMJob(), 8), mLvl);
+
+    uint16 sSTR = mobutils::GetBaseToRank(grade::GetJobGrade(PTrust->GetSJob(), 2), PTrust->GetSLevel());
+    uint16 sDEX = mobutils::GetBaseToRank(grade::GetJobGrade(PTrust->GetSJob(), 3), PTrust->GetSLevel());
```

Comment:
shouldn't you just make `PTrust->GetSLevel()` a variable like you did for mainjob's level to avoid all these repeat calls?